pavelib/prereqs.py

```diff
@@ -103,7 +103,7 @@ def prereq_cache(cache_name, paths, install_func):
     cache_file_path = os.path.join(PREREQS_STATE_DIR, "{}.sha1".format(cache_filename))
     old_hash = None
     if os.path.isfile(cache_file_path):
-        with io.open(cache_file_path, "rb") as cache_file:
+        with io.open(cache_file_path, "r") as cache_file:
```

Comment:
Notes for review. 
The comparison on L#113 fails because of a type mismatch.

Reply:
@awaisdar001 how will this change fix the type mismatch?

Reply:
right now we are reading the `cache_file_path` file as the binary which results in byte type array when you read a file. See L#107

`old_hash = cache_file.read()` # reads as byte string.